### ISSUE TITLE
Import videos from older ICFP workshops

### ISSUE THREAD

**Issue by patricoferris** [ocaml/ocaml.org#112]
There are a few more years that are currently not imported, including 2014, 2012, 2011 and 2010. @tmcgilchrist were you potentially looking into this (re: https://discuss.ocaml.org/t/v3-ocaml-org-a-roadmap-for-ocamls-online-presence/8368/25)? All of the md files should have the YouTube links, although the quality of the videos is... not often great.

**Comment by tmcgilchrist** [comment]
Yes I’m slowly working on this. I’ll have a bit more time over the weekend

**Comment by patricoferris** [comment]
Awesome thank you! Just didn't want to duplicate any work :)) (for bookkeeping this came from #111)

**Comment by tmcgilchrist** [comment]
@patricoferris I'm just missing the 2013 videos from Boston, searching online isn't yielding any links and we don't have anything from https://ocaml.org/meetings/ocaml/2013/

**Comment by patricoferris** [comment]
Yeah I can't find any either, that's okay. I'm guessing they just didn't do videos that year 🤷  (I can't seem to find many ICFP 2013 ones in general)

**Comment by tmcgilchrist** [comment]
Current state of videos:
- [x]  2012 videos uploaded and organised 
- [ ]  2013 missing (ask on community site?)
- [x]  2014 videos uploaded and organised 
- [ ]  2015 videos, I have 7 videos so some of the talks are missing.
- [x]  2016 videos
- [x]  2017 videos
- [x]  2018 videos
- [ ]  2019 videos
- [x]  2020 videos uploaded and organised 
- [x]  2021 videos uploaded and organised 

Also the audio on some of the earlier videos isn't great, when I get more time I'll try and clean it up more.

**Comment by avsm** [comment]
I've gone ahead and imported the videos from 2016, 2017 and 2018 (that are available) to watch.ocaml.org, so they are all in their own channels now. Would someone be able to link them to the website?

@dra27 would you remember why 2018/2019 OCaml workshop videos aren't available online? It's odd that the 2019 ICFP videos are online but not the workshops.

**Comment by dra27** [comment]
I asked back in November - apparently it either wasn't videoed (I confess I can't remember whether there was a camera or not!) or the videos have been lost 😕

**Comment by cuihtlauac** [comment]
I have created one PR for each workshop with videos on watch.ocaml.org that were not already linked from ocaml.org 

* 2012, Copenhagen: https://github.com/ocaml/ocaml.org/pull/752
* 2014, Gothenburg: https://github.com/ocaml/ocaml.org/pull/751
* 2015, Vancouver: https://github.com/ocaml/ocaml.org/pull/750
* 2016, Nara: https://github.com/ocaml/ocaml.org/pull/746
* 2017, Cambridge: https://github.com/ocaml/ocaml.org/pull/747
* 2018, St. Louis: https://github.com/ocaml/ocaml.org/pull/748
* 2022, Ljubljana: https://github.com/ocaml/ocaml.org/pull/749

I tried in each PR to:

1. Add links to watch.ocaml.org, if absent
1. Replace links to YouTube by watch.ocaml.org, whenever possible
1. List missing videos (in the PR description)

**Comment by avsm** [comment]
Thanks for the quick response, @cuihtlauac! All of those PRs look good to me.

**Comment by avsm** [comment]
I did also discover some videos from 2010 on Vimeo: https://vimeo.com/user4287786

However, direct import from Vimeo to YouTube seems broken.  If anyone's able to mirror these to an unlisted YouTube URL, we could then import them into the ocamlmeeting2010 channel on watch.ocaml.org
